### PR TITLE
Add valid value enums to order_by specs

### PR DESCRIPTION
**Pull request by wcmitchell**
Signed-off-by: Chris Mitchell <cmitchel@redhat.com>

## Link(s) to Jira
- https://issues.redhat.com/browse/RHCLOUD-10561

## Description of Intent of Change(s)
Add documentation showing the specific values of order_by filters to openapi spec

## Local Testing
Documentation only, spec generation can be tested by pulling down branch, running `make gen-apidoc`, `make collect-static`, and `make serve` should begin serving the generated spec at /api_prefix/openapi.json

## Checklist
- [x] if API spec changes are required, is the spec updated?

- [ ] are there any pre/post merge actions required? if so, document here.

- [ ] are theses changes covered by unit tests?

- [x] if warranted, are documentation changes accounted for?

- [ ] does this require migration changes?
  - [ ] if yes, are they backwards compatible?

- [ ] is there known, direct impact to dependent teams/components?
  -  if yes, how will this be handled?

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Practices Checklist
- [ ] Input Validation

- [ ] Output Encoding

- [ ] Authentication and Password Management

- [ ] Session Management

- [ ] Access Control

- [ ] Cryptographic Practices

- [ ] Error Handling and Logging

- [ ] Data Protection

- [ ] Communication Security

- [ ] System Configuration

- [ ] Database Security

- [ ] File Management

- [ ] Memory Management

- [ ] General Coding Practices

### CODE REVIEW

File: docs/source/specs/openapi.json [hunk 1]
```diff
@@ -143,6 +143,16 @@
               ],
               "default": "false"
             }
+          },
+          {
+            "in": "query",
+            "name": "order_by",
```
Comment (astrozzc) [inline docs/source/specs/openapi.json:149]: Overall LGTM! Unfortunately, /principals does not support "order_by" yet, we only have a sort_order which works for username only.

Comment (wcmitchell) [reply docs/source/specs/openapi.json:149]: Oh! That's odd, the /groups/uuid/principals/ claims to support it at line 675, should I pull that one out too or does it use a different mechanism?

**Review by coderbydesign — APPROVED** [review]
Generated a client based off these changes, and all calls with `order_by` and the options here, succeeded.

File: docs/source/specs/openapi.json (hunk 1 above)
Comment (astrozzc) [reply docs/source/specs/openapi.json:149]: Right, they have different implementation, /principals/ only supports sorting by username currently, so there is no sort_by option, but sort_order. /groups/uuid/principals/ has order_by implemented without sort_order.

**Review by astrozzc — APPROVED** [review]

**Pull request merged**